include/Mat.h

```diff
@@ -1108,9 +1143,11 @@ class Mat {
         }
         if(strides == NULL) result.buildStrides();
 
-        result.data = data;
-        result.base = result.newMat(result.data);
-        result.base->data = NULL;
+        MatBase<Type>* newBase;
+        newBase = new MatBase<Type>(data);
+        result.base = newBase;
+        result.base->refCount++;
+        result.data = newBase->data;
         return result;
```

Comment:
fiddle with the deallocator system to make it not delete the data here when the base is destroyed